corehq/apps/users/models_sql.py

```diff
@@ -179,6 +184,41 @@ class RoleAssignableBy(models.Model):
     )
 
 
+class UserUpdateMeta(jsonobject.JsonObject):
```

Comment:
why is this a sub model an not just part of the main details object?

Reply:
Sorry, i might have misunderstood how JsonObject is used then.

the SQL model has `details = JSONField(default=dict)`, which would be populated by wrapping using
```
class UpdateDetails(jsonobject.JsonObject):
    meta = jsonobject.ObjectProperty(UserUpdateMeta)
    changes = jsonobject.DictProperty()

class UserUpdateMeta(jsonobject.JsonObject):
    updated_via = jsonobject.StringProperty()
```

The `UserUpdateMeta` is kept separate to allow adding more meta properties in future. May be not a great example but like IP address of the request or user API version.

Reply:
Why can't you just add more properties to the `UpdateDetails` directly? I just don't see the value of pulling it into a nested model.

Reply:
updated to have "updated_via" on UpdateDetails itself.